tests/unit_tests/result_test.py

```diff
@@ -129,13 +129,26 @@ def test_yaml_output(result, mocker):
     """Tests YAML representation of compare result made by YamlOutput class."""
     mocker.patch("diffkemp.output.get_end_line", return_value=2958)
 
-    # mock relpath to return original path
+    old_snapshot_dir = "/abs/path/to/old-snapshot"
+    new_snapshot_dir = "/abs/path/to/new-snapshot"
+
+    # Mocking relpath to return original path, because the file paths
+    # in vertices and etc. in conftest.py are not absolute
+    # (does not contain path to snapshot).
     def mock_rel_path(path, start):
         return path
     mocker.patch("os.path.relpath", side_effect=mock_rel_path)
 
-    yaml = YamlOutput("/abs/path/to/old-snapshot", "/abs/path/to/new-snapshot",
-                      result).output
+    # Mocking path.join for the same reason as relpath.
+    # Note: This mock can be problem if diffkemp uses >2 paths in join.
+    def mock_path_join(path1, *path2):
```

Comment:
Mocking functions from `os.path` sounds very dangerous. Are we not able to come up with a better solution?